src/web/app/pages-instructor/instructor-session-individual-extension-page/instructor-session-individual-extension-page.component.ts

```diff
@@ -0,0 +1,569 @@
+      ...this.feedbackSessionDetails,
+    };
+
+    this.isSubmittingDeadlines = true;
+    this.feedbackSessionsService
+      .updateFeedbackSession(this.courseId, this.feedbackSessionName, request, isNotifyDeadlines)
+      .pipe(finalize(() => { this.isSubmittingDeadlines = false; }))
+      .subscribe(() => {
+          this.loadFeedbackSessionAndIndividuals();
+          this.statusMessageService.showSuccessToast(
+            `Sucessfully created extension(s) for ${selectedStudents.length} student(s)`
+            + ` and ${selectedInstructors.length} instructor(s)!`,
+          );
+        },
+        (resp: ErrorMessageOutput) => {
+          this.statusMessageService.showErrorToast(resp.error.message);
+        },
+      );
+  }
+
+  private getUpdatedDeadlines(
+    selectedIndividuals: StudentExtensionTableColumnModel[] | InstructorExtensionTableColumnModel[],
+    extensionTimestamp: number,
+    isStudent: boolean,
+  ): Record<string, number> {
+    let record: Record<string, number> = {};
+    if (isStudent) {
+      record = { ...this.studentDeadlines };
+    } else {
+      record = { ...this.instructorDeadlines };
+    }
+
+    selectedIndividuals.forEach((x) => {
+      record[x.email] = extensionTimestamp;
+    });
+    return record;
+  }
+
+  private handleDeleteDeadlines(
+    selectedStudents: StudentExtensionTableColumnModel[],
+    selectedInstructors: InstructorExtensionTableColumnModel[],
+    isNotifyDeadlines: boolean,
+  ): void {
+    const request: FeedbackSessionUpdateRequest = {
+      studentDeadlines: this.getDeletedDeadlines(selectedStudents, true),
+      instructorDeadlines: this.getDeletedDeadlines(selectedInstructors, false),
+      ...this.feedbackSessionDetails,
+    };
+
+    this.isSubmittingDeadlines = true;
+    this.feedbackSessionsService
+      .updateFeedbackSession(this.courseId, this.feedbackSessionName, request, isNotifyDeadlines)
+      .pipe(finalize(() => { this.isSubmittingDeadlines = false; }))
+      .subscribe(
+        () => {
+          this.loadFeedbackSessionAndIndividuals();
+          this.statusMessageService.showSuccessToast(
+            `Successfully deleleted extension(s) for ${selectedStudents.length} student(s) and`
+            + ` ${selectedInstructors.length} instructor(s)!`,
+          );
+        },
+        (resp: ErrorMessageOutput) => {
+          this.statusMessageService.showErrorToast(resp.error.message);
+        },
+      );
+  }
+
+  private getDeletedDeadlines(
+    selectedIndividuals: StudentExtensionTableColumnModel[] | InstructorExtensionTableColumnModel[],
+    isStudent: boolean,
+  ): Record<string, number> {
+    let record: Record<string, number> = {};
+    if (isStudent) {
+      record = { ...this.studentDeadlines };
+    } else {
+      record = { ...this.instructorDeadlines };
+    }
+
+    selectedIndividuals.forEach((x) => {
+      delete record[x.email];
+    });
+    return record;
+  }
```

Comment:
Likewise, this is highly duplicated with `getUpdatedDeadlines`. Can we have some abstraction that allow us to reduce the duplicated lines?